state_transfer/transfer/pvc.go

```diff
@@ -0,0 +1,161 @@
+package transfer
+
+import (
+	"crypto/md5"
+	"encoding/hex"
+	"fmt"
+
+	v1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/types"
+)
+
+type PersistentVolumeClaimList []PVC
+
+type PVC interface {
+	Source() PersistentVolumeClaim
+	Destination() PersistentVolumeClaim
+}
+
+type PersistentVolumeClaim interface {
```

Comment:
users will wonder what is the difference between this type and `corev1.PersistentVolumeClaim`

Can we rename this as `LabelSafePersistentVolumeClaim` or `ValidatedPersistentVolumeClaim`? Although the latter could be ambiguous in what the `Validation` is.